### PR TITLE
More track design refactors

### PR DESCRIPTION
**Pull request by Gymnasiast**
- Groups and reorders fields in TrackDesign by category
- Puts the track colours in a more sensible order (mirrorring the `Ride` struct)
- Creates enum classes for colour scheme and vehicle colour settings, removing some duplicated constants in the process

This is done as further preparation for NTDF.

### CODE REVIEW

File: src/openrct2/ride/gentle/SpaceRings.cpp [hunk 1]
```diff
@@ -55,7 +55,7 @@ static void PaintSpaceRingsStructure(
             frameNum += static_cast<int8_t>(vehicle->Pitch) * 4;
         }
 
-        if ((ride.colour_scheme_type & 3) != RIDE_COLOUR_SCHEME_MODE_DIFFERENT_PER_TRAIN)
+        if (ride.vehicleColourSettings != VehicleColourSettings::perTrain)
```
Comment (AaronVanGeffen) [inline src/openrct2/ride/gentle/SpaceRings.cpp:58]: Why was the mask dropped here?

File: src/openrct2/actions/TrackDesignAction.cpp [hunk 2]
```diff
@@ -224,25 +224,22 @@ GameActions::Result TrackDesignAction::Execute() const
...
-    SetOperatingSettingNested(ride->id, RideSetSetting::MinWaitingTime, _td.minWaitingTime, GAME_COMMAND_FLAG_APPLY);
-    SetOperatingSettingNested(ride->id, RideSetSetting::MaxWaitingTime, _td.maxWaitingTime, GAME_COMMAND_FLAG_APPLY);
-    SetOperatingSettingNested(ride->id, RideSetSetting::Operation, _td.operationSetting, GAME_COMMAND_FLAG_APPLY);
-    SetOperatingSettingNested(ride->id, RideSetSetting::LiftHillSpeed, _td.liftHillSpeed & 0x1F, GAME_COMMAND_FLAG_APPLY);
+    SetOperatingSettingNested(ride->id, RideSetSetting::Departure, _td.operation.departFlags, GAME_COMMAND_FLAG_APPLY);
+    SetOperatingSettingNested(ride->id, RideSetSetting::MinWaitingTime, _td.operation.minWaitingTime, GAME_COMMAND_FLAG_APPLY);
+    SetOperatingSettingNested(ride->id, RideSetSetting::MaxWaitingTime, _td.operation.maxWaitingTime, GAME_COMMAND_FLAG_APPLY);
+    SetOperatingSettingNested(ride->id, RideSetSetting::Operation, _td.operation.operationSetting, GAME_COMMAND_FLAG_APPLY);
+    SetOperatingSettingNested(ride->id, RideSetSetting::LiftHillSpeed, _td.operation.liftHillSpeed, GAME_COMMAND_FLAG_APPLY);
```
Comment (AaronVanGeffen) [inline src/openrct2/actions/TrackDesignAction.cpp:239]: Mask was dropped?

**Review by AaronVanGeffen — COMMENTED** [review]
Looks like nice refactor work. Left a few line notes.

File: src/openrct2/ride/RideColour.h [hunk 3]
```diff
@@ -13,7 +13,24 @@
 
 struct TrackColour
 {
-    uint8_t main;
-    uint8_t additional;
-    uint8_t supports;
+    colour_t main;
```
Comment (AaronVanGeffen) [inline src/openrct2/ride/RideColour.h:16]: Needs a `common.h` header import. (Though we should probably move the type definition over to `Colour.h`.)

File: src/openrct2/ride/gentle/SpaceRings.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/ride/gentle/SpaceRings.cpp:58]: Nothing else is saved in this field, so I was able to change its type to the enum class, which makes the mask superfluous.

File: src/openrct2/actions/TrackDesignAction.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/actions/TrackDesignAction.cpp:239]: That is correct, it is no longer necessary. In the past, the lift hill speed field had the number of circuits in the upper 3 bits, but that has been split in a previous refactor. I just forgot to remove the mask back then.

File: src/openrct2/ride/RideColour.h (hunk 3 above)
Comment (Gymnasiast) [reply src/openrct2/ride/RideColour.h:18]: Done. Could you create an issue for the latter?

**Review by ZeeMaji — APPROVED** [review]
Tested importing 25 various td4/td6 designs, along with saving 5 back to td6. Everything seems to be working as expected.

**Pull request merged**